shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/config/datasource/ShardingSphereJDBCDataSourceCreator.java

```diff
@@ -36,8 +38,19 @@
     @Override
     public DataSource createDataSource(final Object dataSourceConfig) throws SQLException {
         YamlRootConfiguration rootConfig = (YamlRootConfiguration) dataSourceConfig;
+        ShardingRuleConfiguration shardingRuleConfig = ShardingRuleConfigurationConverter.findAndConvertShardingRuleConfiguration(rootConfig.getRules());
+        enableRangeQueryForInline(shardingRuleConfig);
         return ShardingSphereDataSourceFactory.createDataSource(rootConfig.getSchemaName(), new YamlDataSourceConfigurationSwapper().swapToDataSources(rootConfig.getDataSources()), 
-                Collections.singletonList(ShardingRuleConfigurationConverter.findAndConvertShardingRuleConfiguration(rootConfig.getRules())), null);
+                Collections.singletonList(shardingRuleConfig), null);
+    }
+    
+    private void enableRangeQueryForInline(final ShardingRuleConfiguration shardingRuleConfig) {
+        for (ShardingSphereAlgorithmConfiguration each : shardingRuleConfig.getShardingAlgorithms().values()) {
+            if (!"INLINE".equalsIgnoreCase(each.getType())) {
+                continue;
+            }
+            each.getProps().put("allow-range-query-with-inline-sharding", "true");
```

Comment:
maybe this should be a constant ?

Reply:
I tried to find the `allow-range-query-with-inline-sharding` constant definition, not defined in infra.